### PR TITLE
FEATURE: Iterable multi cache backend

### LINKED ISSUE #2894: FEATURE: IterableMultiBackend
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

In order to store sessions in caches controlled by a Multi Backend, there needs to be an implementation which supports the `IterableBackendInterface`.

### Possible Solution

Implement a specialized version of the `TaggableMultiBackend` which implements the `IterableBackendInterface`.

### PR DESCRIPTION
**Pull request by robertlemke**
This introduces a new cache backend, based on the `TaggableMultiBackend`, which can also be used for caches which require iteration support.

Resolves #2894

### CODE REVIEW

**Comment by robertlemke** [comment]
Happy Psalm, happy Robert!

File: Neos.Cache/Classes/Backend/IterableMultiBackend.php [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+     * @throws Throwable
+     */
+    protected function buildSubBackend(string $backendClassName, array $backendOptions): ?BackendInterface
+    {
+        if (!is_subclass_of($backendClassName, IterableBackendInterface::class)) {
+            $message = sprintf('Failed building sub backend %s for %s because it does not implement %s', $backendClassName, get_class($this), IterableBackendInterface::class);
+            $throwable = new Exception($message);
+            $this->logger?->error(($this->throwableStorage ? $this->throwableStorage->logThrowable($throwable) : $message), LogEnvironment::fromMethodName(__METHOD__));
+            $this->handleError($throwable);
+            return null;
```
Comment (kdambekalns) [inline Neos.Cache/Classes/Backend/IterableMultiBackend.php:35]: I like that, see https://github.com/neos/flow-development-collection/pull/2896

**Review by kdambekalns — APPROVED** [review]

**Comment by robertlemke** [comment]
Thanks!

**Review by kitsunet — APPROVED** [review]
seems fine, iterating might be a bit wonky if a backend fails and it "continues" on the next but not much we cam do

**Pull request merged**